modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/PlaybackManager.kt

```diff
@@ -960,26 +960,38 @@ open class PlaybackManager @Inject constructor(
                     }
                 }
             }
-
-            playbackStateRelay.blockingFirst().let { playbackState ->
-                val updatedItems = playbackState.chapters.getList().map {
-                    if (it.index == chapter.index) {
-                        it.copy(selected = select)
-                    } else {
-                        it
+            withContext(Dispatchers.Main) {
+                playbackStateRelay.blockingFirst().let { playbackState ->
```

Comment:
Is it necessary to switch to the `Main` dispatcher here? If it is `blockingFirst()` will block the main thread. It can be avoided with `playbackStateRelay.firstOrError().await()`.

Reply:
I noticed in one instance that toggling was reverted without switching to the main dispatcher. I can't reproduce it now. I have removed the switch but I'll continue testing it.